### PR TITLE
Fix freeze projectiles only hit one tee (Fixes #1759)

### LINKED ISSUE #1759: Freeze projectile bug
If two tees hit a freezing bullet on the same tick, (can reproduce in a solo part using dummy copy move) only one of them gets frozen.

This seems to be because we just do `IntersectCharacter` in `projectile.cpp` which returns only one of the tees.

### PR DESCRIPTION
**Pull request by fokkonaut**

### CODE REVIEW

File: src/game/server/entities/projectile.cpp [hunk 1]
```diff
@@ -168,8 +168,14 @@ void CProjectile::Tick()
...
 			}
 		}
-		else if(pTargetChr && m_Freeze && ((m_Layer == LAYER_SWITCH && GameServer()->Collision()->m_pSwitchers[m_Number].m_Status[pTargetChr->Team()]) || m_Layer != LAYER_SWITCH))
-			pTargetChr->Freeze();
+		else if(m_Freeze)
+		{
+			CCharacter *apEnts[MAX_CLIENTS];
+			int Num = GameWorld()->FindEntities(CurPos, 1.0f, (CEntity**)apEnts, MAX_CLIENTS, CGameWorld::ENTTYPE_CHARACTER);
+			for(int i = 0; i < Num; ++i)
+				if(apEnts[i] && (m_Layer != LAYER_SWITCH || (m_Layer == LAYER_SWITCH && GameServer()->Collision()->m_pSwitchers[m_Number].m_Status[pTargetChr->Team()])))
```
Comment (fokkonaut) [inline src/game/server/entities/projectile.cpp]: Has to be `apEnts[i]->Team()`, will do that tomorrow

**Comment by def-** [comment]
This is a physics change. Are we sure we want this?

**Comment by fokkonaut** [comment]
This is an even smaller bug than the Grenade Double Explosion, which got fixed. And this one can be abused in Solomaps, and there is no map which uses this bug as a part, so there should be no reason against merging.

**Comment by def-** [comment]
Ok, fine with me. @Learath2 @heinrich5991 ?

**Comment by heinrich5991** [comment]
Fine.

**Comment by trml** [comment]
Are you sure FindEntities will work the same as IntersectCharacter? FindEntities seems like it searches around one point while IntersectCharacter searches along (around) a line.

**Comment by fokkonaut** [comment]
i've tested it and it seems to work fine

**Comment by trml** [comment]
Ok. Don't know if it would be noticeable in practice since bullets move slowly, but seems like it could potentially lead to a slightly smaller freeze area in some rare cases.

**Comment by Learath2** [comment]
This could be changing whether you can sit between two bullets without getting frozen or not

**Comment by Learath2** [comment]
@fokkonaut I think this should be done with `CGameWorld::IntersectedCharacters` instead

**Pull request merged**